lib/imparse.c

```diff
@@ -214,3 +215,39 @@ EXPORTED int imparse_isnumber(const char *s)
     }
     return 1;
 }
+
+/*
+ * Parse a range from the string starting at the pointer pointed to by 's'.
+ * and populate the structure in the pointer at 'range'.
+ * Returns 0 on success, and non-zero on failure.
+ */
+EXPORTED int imparse_range(char *s, range_t *range)
+{
+    if (*s == '-') {
+        range->is_last = 1;
+        s++;
+    }
+    else if (!Uisdigit(*s)) return -1;
+
+    errno = 0;
+    range->low = strtoul(s, &s, 10);
+    if (!range->low || errno || *s != ':') return -1;
```

Comment:
Since we're handling `errno` here (by detecting it and returning -1) we should reset it back to zero when we do, so that it doesn't add irrelevant logging noise later

```suggestion
    if (!range->low || errno || *s != ':') {
        errno = 0;
        return -1;
    }
```

Reply:
done